### PR TITLE
memory: increase hpsram buffer size for CNL

### PR DESCRIPTION
**Pull request by mrajwa**
This patch increases HPSRAM buffer size. This
additional space is needed for KWD functionality.
Also, this patch changes HP buffer block size to
improve allocation granularity.

Signed-off-by: Marcin Rajwa <marcin.rajwa@linux.intel.com>

### CODE REVIEW

**Comment by jajanusz** [comment]
@mrajwa I know we don't have something like that atm in fw, but I'd like to increase heap size only when some components are used. F.e. like here when CONFIG_KWD is enabled then add some heap size.

**Comment by mrajwa** [comment]
Updated

**Comment by tlauda** [comment]
SOFCI TEST

File: src/platform/cannonlake/include/platform/lib/memory.h [hunk 1]
```diff
@@ -215,8 +215,12 @@
 #define HP_SRAM_WIN3_SIZE	SRAM_TRACE_SIZE
 
 #define HEAP_HP_BUFFER_BASE		(SRAM_TRACE_BASE + SRAM_TRACE_SIZE)
+#if CONFIG_COMP_KPB
+#define HEAP_HP_BUFFER_SIZE		0x40000
+#else
 #define HEAP_HP_BUFFER_SIZE		0x20000
-#define HEAP_HP_BUFFER_BLOCK_SIZE	0x180
```
Comment (jajanusz) [inline src/platform/cannonlake/include/platform/lib/memory.h]: Why did you change block size?

Comment (mrajwa) [reply src/platform/cannonlake/include/platform/lib/memory.h]: Because, as described in commit message smaller blocks allows more accurate allocations.

Comment (jajanusz) [reply src/platform/cannonlake/include/platform/lib/memory.h]: ok, thanks
